File: samples/search/php/perform-search.php
<?php

$curl = curl_init();

$apiKey = "";    // Replace with your API Key
$trackID = "";  // Replace with track ID
$searchText = ""; // Replace with search text
$page = 0;  // Replace with page number
$items = 50;  // Replace with items per page

$requestBody = json_encode( array( "id"=> $trackID, "text"=> $text, "page"=> $page, "items"=> $items) );
$authHeader = $apiKey.":"."";
$encodedAuthHeader = base64_encode($authHeader);

curl_setopt_array($curl, array(
  CURLOPT_URL => "https://api-us.musiio.com/v1/search/perform",
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => "",
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => "POST",
  CURLOPT_POSTFIELDS =>$requestBody,
  CURLOPT_HTTPHEADER => array(
    "Content-Type: application/json",
    "Authorization: Basic ".$encodedAuthHeader
  ),
));

$response = curl_exec($curl);

curl_close($curl);
echo $response;
?>

File: samples/search/java/PerformSearch.java
package com.example.search;    // Replace it with your own project group ID

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

import org.springframework.http.HttpEntity;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.web.client.RestTemplate;

public class PerformSearch {
    private static final String API_KEY = "";   // Your API Key here
    private static final String URL = "https://api-us.musiio.com/v1/search/perform";

    public static void main(String[] args) {
        String trackId = args[0]; // input track ID in command: "-Dexec.args='track id here...'"
        String searchText = "";
        int page = 0;
        int items = 50;
        try {
            searchText = args[1];
            page = Integer.parseInt(args[2]);
            items = Integer.parseInt(args[3]);
        } catch (NumberFormatException e) {
            System.err.println("Please enter integer values for 3rd and 4th arguments if needed");
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Use default values for optional arguments");
        }
        HttpHeaders headers = HeadersUtils.createHeaders(API_KEY, "");
        headers.setContentType(MediaType.APPLICATION_JSON);
        RestTemplate restTemplate = new RestTemplate();
        ObjectMapper mapper = new ObjectMapper();
        ObjectNode requestBody = mapper.createObjectNode();
        requestBody.put("id", trackId);
        requestBody.put("text", searchText);
        requestBody.put("page", page);
        requestBody.put("items", items);
        HttpEntity<String> request = new HttpEntity<>(requestBody.toString(), headers);
        String response = restTemplate.postForObject(URL, request, String.class);
        System.out.println(response);
    }
}
